scripts/screeps_rl_scorecard.py

```
@@ -254,6 +276,86 @@ def normalized_key(value: str) -> str:
 
+def iter_json_objects(value: Any) -> Iterable[JsonObject]:
+    if isinstance(value, dict):
+        yield value
+        for item in value.values():
+            yield from iter_json_objects(item)
+    elif isinstance(value, list):
+        for item in value:
+            yield from iter_json_objects(item)
+
+
+def value_has_reference(value: Any) -> bool:
```

Ruff (0.15.13)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)